### DISCUSSION TITLE
Bots are trapped in reactor room of EngineeringModule_01

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15061] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

![20241105123227](https://github.com/user-attachments/assets/3b0126f3-79b6-4499-b8fd-956641061519)
Bots were keeping climbing up and down

### Reproduction steps

1. Test EngineeringModule_01 in submarine editor
2. teleport bots inside reactor room
3. they are likely unable to leave

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
https://github.com/user-attachments/assets/9d1f32a4-548a-49a8-871c-85c8f255687d

it only happens when bots are trying to climb the ladder from left side

**Comment by itchyOwl** [comment]
Hmm, I can't repro it in the sub editor test mode. Could you enable `debugai`, which shows the objectives and the path.

**Comment by A2ur3** [comment]
https://github.com/user-attachments/assets/473d34c1-b45c-4756-811f-597fa133a51f

**Comment by itchyOwl** [comment]
I was just about to write that still can't repro, but then I noticed that I catched it.  Seems to only happen with bots that have `StayInHull` behavior (e.g. captains have that), which are set to target the current hull while they are climbing. So they try to navigate to the room they are already in. That's why they flip the orientation in the ladders (because they target the center of the room). So the bots are not actually stuck there. They are in the idle state, but unable to get off from the ladders in that state. So, while it's not a severe issue, it looks very glitchy. I'll try to fix it.

**Comment by itchyOwl** [comment]
Addressed in FakeFishGames/Barotrauma-development@e7ca5d0.

**Comment by Regalis11** [comment]
Thank you for the report! The issue has been addressed in our private repo now, fix coming up in a future unstable build.
